### PR TITLE
docs: update `CONTRIBUTING.md`

### PR DESCRIPTION
**Pull request by angeloashmore**
**Resolves**: N/A 

### Description

This PR replaces the existing `CONTRIBUTING.md` with a new, simplified version.

### Checklist

- [ ] If my changes require tests, I added them.
- [ ] If my changes affect backward compatibility, it has been discussed.
- [x] If my changes require an update to the CONTRIBUTING.md guide, I updated it.

### Preview

- https://github.com/prismicio/prismic-client/blob/aa/docs/CONTRIBUTING.md

### How to QA [^1]

Read through `CONTRIBUTING.md` and let me know what you think.

[^1]:
    Please use these labels when submitting a review:
    :question: #ask:&ensp;Ask a question.
    :bulb: #idea:&ensp;Suggest an idea.
    :warning: #issue:&ensp;Strongly suggest a change.
    :tada: #nice:&ensp;Share a complement.

### CODE REVIEW

**Review by lihbr — APPROVED** [review]
I really like it! Noticed a small typo

What do you think about just deleting the Pull Request template from here and move it to a `.github` repository already?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,448 +1,137 @@
...
 
-To release a new version of the project:
+```sh
+# Checkout the master branch and pull the latest changes.
+git checkout master
+git pull
 
-```bash
+# Perform a dry-run and verify the output.
+# If it looks good, release a new # version.
```
Comment (lihbr) [inline CONTRIBUTING.md:123]: Not sure I understand `#` in `new # version` here and below 🤔

Comment (angeloashmore) [reply CONTRIBUTING.md]: Oops, it's a typo. It probably came from removing a line break that started with a `#`.

**Comment by angeloashmore** [comment]
The typos were fixed and `PULL_REQUEST_TEMPLATE.md` was removed in favor of the org-level version. We should be good to merge!

**Pull request merged**